### PR TITLE
More docs improvements

### PR DESCRIPTION
**Pull request by pfrazee**
- Reorganize spec documents into (hopefully) the final structure of 5 main specs
  - Removed old spec-generator code
- Add new specs:
  - FedRPC
  - NSID
  - ADX Name Resolution
  - ADX FedRPC Methods
- Remove old specs:
  - HTTP routes
- Improve existing specs:
  - Add schemas to ADX Repo
  - Add ID forms to ADX Repo
- Still todo:
  - did:pch
  - DID Resolution spec
  - ADX Repo structure & encoding
  - FedRPC authentication model
  - All FedRPC method schemas
  - Final names and IDs of protocols and applications
  - The social application's fedrpc and adx record schemas

### CODE REVIEW

File: docs/specs/nsid.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# NameSpaced IDs (NSIDs)
```
Comment (dholms) [inline docs/specs/nsid.md:1]: thoughts on adding a note to this about what we discussed this morning re: no implied hierarchy with sub domains?

File: docs/specs/did-pch.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# DID Placeholder (did:pch)
+
+TODO
```
Comment (dholms) [inline docs/specs/did-pch.md:3]: 🏃‍♂️

File: docs/specs/adx/repo.md [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+### Schema validation
+
+Constraints are structural: they apply constraints to fields under object path (eg `#/text`) to establish permissable values for that field. The constraints they can apply are value-type and valid values of the type (eg numbers within a range, strings of a certain format or pattern, etc). These constraints are described using [JSON Schema](https://json-schema.org/draft/2020-12/json-schema-core.html).
+
+Unconstrained fields are ignored during validation, but should be avoided in case future versions of the schema apply constraints.
+
+### Schema versioning
+
+Once a field constraint is published, it can never change. Loosening a constraint (adding possible values) will cause old software to fail validation for new data, and tightening a constraint (removing possible values) will cause new software to fail validation for old data. As a consequence, schemas may only add constraints to previously unconstrained fields.
```
Comment (dholms) [inline docs/specs/adx/repo.md:196]: > schemas may only add constraints to previously unconstrained fields.

wouldn't this be "tightening a constraint?"

**Review by dholms — APPROVED** [review]
a couple of small comments, but over all looks really good 🙌

File: docs/specs/nsid.md (hunk 1 above)
Comment (pfrazee) [reply docs/specs/nsid.md:1]: Done

File: docs/specs/adx/repo.md (hunk 3 above)
Comment (pfrazee) [reply docs/specs/adx/repo.md:196]: Technically yes but we ignore unconstrained fields as a kind of special case, and if you do this then the new constraint has to be optional. I'll make sure that's noted.

**Pull request merged**